website/server/libs/user/index.js

```diff
@@ -112,7 +116,68 @@ export async function update (req, res, { isV3 = false }) {
     const newName = req.body['profile.name'];
     if (newName === null) throw new BadRequest(res.t('invalidReqParams'));
     if (newName.length > 30) throw new BadRequest(res.t('displaynameIssueLength'));
-    if (nameContainsSlur(newName)) throw new BadRequest(res.t('displaynameIssueSlur'));
+    if (stringContainsSlur(newName)) {
+      user.flags.chatRevoked = true;
+      await user.save();
+
+      // Email the mods
+      const authorEmail = getUserInfo(user, ['email']).email;
+
+      const report = [
+        { name: 'PROFILE_NAME_CHANGE_TIME', content: (new Date()).toString() },
+        { name: 'PROFILE_NAME_CHANGE_TEXT', content: newName },
+
+        { name: 'AUTHOR_USERNAME', content: user.profile.name },
+        { name: 'AUTHOR_UUID', content: user._id },
+        { name: 'AUTHOR_EMAIL', content: authorEmail },
+        { name: 'AUTHOR_MODAL_URL', content: `/profile/${user._id}` },
+      ];
+
+      sendTxn(FLAG_REPORT_EMAILS, 'slur-report-to-mods', report);
+
+      // Slack the mods
+      slack.sendSlurNotification({
+        authorEmail,
+        author: user,
+        message: `Name change: "${newName}"`,
+      });
+
+      throw new BadRequest(res.t('displaynameIssueSlur'));
+    }
+    if (stringContainsBannedWord(newName)) throw new BadRequest(res.t('displaynameIssueSlur'));
+  }
+
+  if (req.body['profile.blurb'] !== undefined) {
```

Comment:
here I'd check for `req.body['profile.blurb'] != null` which will cover both null and undefined, usually I'd check directly for `if (req.body['profile.blurb'])` but that would not allow the user to remove the blurb once set